tensorflow/compiler/mlir/tosa/tests/tfl-to-tosa-pipeline.mlir

```diff
@@ -1278,3 +1278,15 @@ func @test_fullyconnected_hybrid(%arg0: tensor<14x19xf32>) -> tensor<14x28xf32>
   %2 = "tfl.fully_connected"(%arg0, %0, %1) {fused_activation_function = "NONE", keep_num_dims = false, weights_format = "DEFAULT"} : (tensor<14x19xf32>, tensor<28x19x!quant.uniform<i8:f32, 1.0>>, tensor<28xf32>) -> tensor<14x28xf32>
   return %2 : tensor<14x28xf32>
 }
+
+// -----
+
+func @test_conv2d_infer(%arg0: tensor<1x32x32x8xf32>, %arg1: tensor<16x1x1x8xf32>) -> tensor<*xf32> {
+  %cst = constant dense<0.000000e+00> : tensor<16xf32>
+  // CHECK: tosa.add
+  // CHECK: tosa.conv2d
+  // CHECKL tensor.cast
```

Comment:
Type: 'CHECKL' -> 'CHECK:'

Can you extend this test? I am guessing that it is supposed to be updating the shape of the: a) conv result, b) return, c) function signature.

Probably fine to verify a&b since the verifier will enforce c.

Reply:
Also, looking through the rest of this patch, a lot of ops are touched but only this one test. Wdyt of strengthening the corresponding tests above by changing them to dynamic shaped versions and then making the existing CHECKs more specific to verify the shape. A lot of the tests in this file are a bit weak to my eye, and this could split the difference of improving some of them and adding test coverage for this PR in one go.

Reply:
Updated tests and lowering where possible to have unknown output shape. There is additional work
required for the rest but this triages down to a select few.

Tests may be failing, working on cleaning up checks.